src/anm/analytics.js

```diff
@@ -0,0 +1,79 @@
+var engine = require('engine'),
+    utils = require('./utils.js');
+
+var Analytics = function () {
+    var self = this,
+        supportSendBeacon = !!navigator.sendBeacon,
+        timeout = 1000,
+        beacon = null,
+        animatronUrl = utils.makeApiUrl('/analytics/player');
+
+    self.enabled = animatronUrl != null && animatronUrl.indexOf('animatron-test') >= 0;
+    self.queue = [];
+
+    var event = function () {
+        if (self.queue.length > 0) {
+            var data = JSON.stringify(self.queue);
+            self.queue = [];
+
+            if (supportSendBeacon) {
+                navigator.sendBeacon(animatronUrl, data);
+            } else {
+                sendViaGif(data);
+            }
+        }
+        if (!supportSendBeacon) {
+            setTimeout(event, timeout);
+        }
+    };
+
+    var sendViaGif = function (data) {
+        if (!beacon) {
+            beacon = engine.createStatImg();
+        }
+        beacon.src = animatronUrl + '?data=' + encodeURIComponent(data);
+    };
+    if (self.enabled) {
+        if (!supportSendBeacon) {
+            engine.onDocReady(event);
```

Comment:
Omg, Anton. Like 10 times I already told how it works.
'Safety' was about running the event loop for sendBeacon. I removed it.

Now it is performed only one time if browser supports sendBeacon. But not all of them support it. So I need to send them every N seconds in such case.


And now, pls tell me what are benefits of starting event loop when player is loaded? That is what I call premature optimizations. Not checking for length of an array. It just saves nothing. Absolutely nothing. But it does indeed complicates code. For nothing. Can you understand my point now? Because you either don't get what I am saying or understand it differently. I have to tell the same things many times, again and again.